app-backend/tool/src/main/scala/ast/package.scala

```diff
@@ -87,4 +87,16 @@ package object ast extends MapAlgebraCodec {
     def min(other: MapAlgebraAST): MapAlgebraAST.Operation =
       MapAlgebraAST.Min(List(self, other), UUID.randomUUID(), generateMetadata)
   }
+
+  implicit class FocalBuffer(val self: MapAlgebraAST) {
```

Comment:
I figured that we're going to be adding a lot of this sort of thing soon and that the trait was getting pretty polluted. This also allowed me to keep all the recursion neatly packed in one spot